chat/uikit/android.md

```diff
@@ -0,0 +1,246 @@
+
+
+#### Event Hooks
+UIKit provides event hooks which enable you to implement specific feature when an event is fired.
+
+The below code segment passes a listener to a `ConversationActivity`.
+
+```kotlin
+i.putExtra(ConversationActivity.MessageFetchListenerIntentKey, object: MessageFetchListener {
+    override fun onBeforeMessageFetch(fragment: ConversationFragment) {
+
+    }
+
+    override fun onMessageFetchFailed(fragment: ConversationFragment, error: Error) {
+        Toast.makeText(fragment.activity, "Message Loading Failed", Toast.LENGTH_LONG).show()
+    }
+
+    override fun onMessageFetchSuccess(fragment: ConversationFragment, messages: List<Message>, isCached: Boolean) {
+        Toast.makeText(fragment.activity, "Message Loaded", Toast.LENGTH_LONG).show()
+    }
+})
+```
+
+
+
+##### `MessageSentListener`
+`MessageSentListener` is an interface for callbacks to be involved when a message is sent.
+
+- `onBeforeMessageSent` is called before a message is sent to Skygear Chat server.
+- `onMessageSentSuccess` is called after a message is sent.
+- `onMessageSentFailed` is called after a message is failed.
+
+
+##### `MessageFetchListener`
+`MessageFetchListener` is an interface for callbacks to be involved when messages of a conversation are fetched.
+
+- `onBeforeMessageFetch ` is called before messages are fetched.
+- `onMessageFetchSuccess` is called after messages are fetched.
+- `onMessageFetchFailed ` is called after fetching is failed.
+
+##### `ConnectionListener`
+`ConnectionListener` is an interface for callbacks to be involved when connection to server is changed.
+
+- `onClose` is called when pubsub connection is closed.
+- `onOpen` is called when pubsub connection is established.
+
+##### Intent Keys for each type of listeners
+
+|Key|Listener|
+|---|--------|
+|`MessageSentListenerIntentKey`|MessageSentListener|
+|`MessageFetchListenerIntentKey`|MessageFetchListener|
+|`ConnectionListenerIntentKey`|ConnectionListener|
+
+
+#### Title Option
+UIKit provides two title options
+
+1. `DEFAULT` and
+2. `OTHER_PARTICIPANTS`
+
+If `DEFAULT` is set, then the title bar displays `title` field value of the conversation record. If  `OTHER_PARTICIPANTS` is set,
+then the title bar displays participants' usernames (excluding the current user) in the title bar.
+
+```java
+Intent intent = new Intent(this, ConversationActivity.class);
+intent.putExtra(ConversationActivity.ConversationIntentKey, c.toJson().toString());
+intent.putExtra(ConversationActivity.TitleOptionIntentKey, ConversationTitleOption.OTHER_PARTICIPANTS);
+startActivity(intent);
+```
+
+#### Adapter
+UIKit provides `ConversationViewAdapter` to enable remote HTTP background image in `ConversationActivity`.
+
+```java
+ConversationViewAdapter adapter = new ConversationViewAdapter() {
+    @Override
+    public String backgroundImageURL(Conversation conversation) {
+        return "https://skygear.io/static/images/skygear-logo.png";
+    }
+};
+```
+### Avatar
```

Comment:
Should this be included in the attribute customisation > avatar? So I don't have to scroll down to bottom in order to learn about everything about avatar.

Reply:
I think we better keep avatar away from attribute customisation because Avatar section would focuses on customisable code/interface while attribute customisation focuses on setting things in intent or XML. So I would rename it as *Avatar Adapter* instead.